src/EA.Weee.Web.Tests.Unit/Areas/Scheme/Controllers/TransferEvidenceControllerTests.cs

```diff
@@ -286,6 +319,227 @@ public async Task TransferEvidenceNotePost_GivenInvalidModel_SessionShouldNotBeE
+                    SessionKeyConstant.TransferNoteKey)).MustHaveHappenedOnceExactly();
+        }
+
+        [Fact]
+        public async Task TransferFromGet_GivenTransferNoteSessionObjectIsRetrievedAndIsNull_ArgumentNulLExceptionExpected()
+        {
+            //arrange
+            A.CallTo(() =>
+                sessionService.GetTransferSessionObject<TransferEvidenceNoteRequest>(transferEvidenceController.Session,
+                    SessionKeyConstant.TransferNoteKey)).Returns(null);
+
+            // act
+            var result = await Record.ExceptionAsync(async () => await transferEvidenceController.TransferFrom(organisationId));
+
+            // assert
+            result.Should().BeOfType<ArgumentNullException>();
+        }
+
+        [Fact]
+        public async Task TransferFromGet_GivenTransferNoteSessionObject_TransferNotesShouldBeRetrieved()
+        {
+            //arrange
+            var request = GetRequest();
+            A.CallTo(() =>
+                sessionService.GetTransferSessionObject<TransferEvidenceNoteRequest>(transferEvidenceController.Session,
+                    SessionKeyConstant.TransferNoteKey)).Returns(request);
+
+            // act
+            await transferEvidenceController.TransferFrom(organisationId);
+
+            // assert
+            A.CallTo(() => weeeClient.SendAsync(A<string>._,
+                    A<GetEvidenceNotesForTransferRequest>.That.Matches(g =>
+                        g.Categories.Equals(request.CategoryIds) && g.OrganisationId.Equals(organisationId))))
+                .MustHaveHappenedOnceExactly();
+        }
+
+        [Fact]
+        public async Task TransferFromGet_GivenTransferNoteSessionObjectAndNotes_ModelShouldBeMapped()
+        {
+            //arrange
+            var request = GetRequest();
+            var notes = fixture.CreateMany<EvidenceNoteData>().ToList();
+
+            A.CallTo(() =>
+                sessionService.GetTransferSessionObject<TransferEvidenceNoteRequest>(transferEvidenceController.Session,
+                    SessionKeyConstant.TransferNoteKey)).Returns(request);
+            A.CallTo(() => weeeClient.SendAsync(A<string>._,
+                A<GetEvidenceNotesForTransferRequest>._)).Returns(notes);
+
+            // act
+            await transferEvidenceController.TransferFrom(organisationId);
+
+            // assert
+
+            A.CallTo(() => mapper.Map<TransferEvidenceNotesViewModelMapTransfer, TransferEvidenceNotesViewModel>(
+                A<TransferEvidenceNotesViewModelMapTransfer>.That.Matches(t =>
+                    t.OrganisationId.Equals(organisationId) &&
+                    t.Notes.Equals(notes) &&
+                    t.Request.Equals(request)))).MustHaveHappenedOnceExactly();
+        }
+
+        [Fact]
+        public async Task TransferFromGet_GivenTransferNoteSessionObjectAndNotes_ModelShouldBeReturned()
```

Comment:
update the name to WithValidModel or something, i think it's copy pasted from above,

Reply:
Updated